<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Unified access to DNA methylation based cell-type deconvolution methods • methyldeconv</title>
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="deps/headroom-0.11.0/headroom.min.js"></script><script src="deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="deps/search-1.0.0/fuse.min.js"></script><script src="deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="Unified access to DNA methylation based cell-type deconvolution methods">
<meta name="description" content="runs several cell type deconvolution algorithms.">
<meta property="og:description" content="runs several cell type deconvolution algorithms.">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="index.html">methyldeconv</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.3</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="reference/index.html">Reference</a></li>
<li class="nav-item"><a class="nav-link" href="news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/omnideconv/methyldeconv/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header"><h1 id="methyldeconv">methyldeconv<a class="anchor" aria-label="anchor" href="#methyldeconv"></a>
</h1></div>
<!-- badges: start -->

<!-- badges: end -->
<p>Ever wanted to apply cell-type deconvolution on your DNA methylation data but could not decide which method to use? Here is <strong>methyldeconv</strong> to help your needs!</p>
<p>This package integrates unified access to five reference-based cell-type deconvolution methods that can directly be applied to Illumina array data (450k, EPIC arrays) or bisulfite sequencing data (RRBS, WGBS).</p>
<p>The included methods are:</p>
<table class="table">
<colgroup>
<col width="33%">
<col width="33%">
<col width="33%">
</colgroup>
<thead><tr class="header">
<th>method</th>
<th>license</th>
<th>citation</th>
</tr></thead>
<tbody>
<tr class="odd">
<td><a href="https://bioconductor.org/packages/release/bioc/html/EpiDISH.html" class="external-link">EpiDISH</a></td>
<td>GPL-2</td>
<td>Teschendorff, A.E., Breeze, C.E., Zheng, S.C. <em>et al.</em> A comparison of reference-based algorithms for correcting cell-type heterogeneity in Epigenome-Wide Association Studies. <em>BMC Bioinformatics</em> <strong>18</strong>, 105 (2017). <a href="https://doi.org/10.1186/s12859-017-1511-5" class="external-link uri">https://doi.org/10.1186/s12859-017-1511-5</a>
</td>
</tr>
<tr class="even">
<td><a href="https://www.bioconductor.org/packages/release/data/experiment/html/FlowSorted.Blood.EPIC.html" class="external-link">Houseman (Flow.Sorted.Blood)</a></td>
<td>GPL-3</td>
<td>Houseman, E.A., Accomando, W.P., Koestler, D.C. <em>et al.</em> DNA methylation arrays as surrogate measures of cell mixture distribution. <em>BMC Bioinformatics</em> <strong>13</strong>, 86 (2012). <a href="https://doi.org/10.1186/1471-2105-13-86" class="external-link uri">https://doi.org/10.1186/1471-2105-13-86</a> Koestler, D.C., Jones, M.J., Usset, J. et al. Improving cell mixture deconvolution by identifying optimal DNA methylation libraries (IDOL).BMC Bioinformatics 17, 120 (2016). <a href="https://doi.org/10.1186/s12859-016-0943-7" class="external-link uri">https://doi.org/10.1186/s12859-016-0943-7</a> Salas, L.A., Koestler, D.C., Butler, R.A. et al. An optimized library for reference-based deconvolution of whole-blood biospecimens assayed using the Illumina HumanMethylationEPIC BeadArray. Genome Biol 19, 64 (2018). <a href="https://doi.org/10.1186/s13059-018-1448-7" class="external-link uri">https://doi.org/10.1186/s13059-018-1448-7</a>
</td>
</tr>
<tr class="odd">
<td><a href="https://github.com/nloyfer/meth_atlas" class="external-link">MethAtlas</a></td>
<td>
<a href="https://github.com/nloyfer/meth_atlas/blob/master/LICENSE.md" class="external-link">Yissum Software Research License</a> “This software is distributed under the <strong>Yissum Software Research License</strong>, which permits use, modification, and distribution for <strong>research purposes only</strong>. Commercial use requires a separate license from Yissum (software@yissum.co.il).”</td>
<td>Moss, J., Magenheim, J., Neiman, D. <em>et al.</em> Comprehensive human cell-type methylation atlas reveals origins of circulating cell-free DNA in health and disease. <em>Nat Commun</em> <strong>9</strong>, 5068 (2018). <a href="https://doi.org/10.1038/s41467-018-07466-6" class="external-link uri">https://doi.org/10.1038/s41467-018-07466-6</a>
</td>
</tr>
<tr class="even">
<td><a href="https://github.com/stephaniehicks/methylCC" class="external-link">methylCC</a></td>
<td>GPL-3</td>
<td>Hicks, S.C., Irizarry, R.A. methylCC: technology-independent estimation of cell type composition using differentially methylated regions. <em>Genome Biol</em> <strong>20</strong>, 261 (2019). <a href="https://doi.org/10.1186/s13059-019-1827-8" class="external-link uri">https://doi.org/10.1186/s13059-019-1827-8</a>
</td>
</tr>
<tr class="odd">
<td><a href="https://github.com/darneson/MethylResolver" class="external-link">methylResolver</a></td>
<td>GPL-3</td>
<td>Arneson, D., Yang, X. &amp; Wang, K. MethylResolver—a method for deconvoluting bulk DNA methylation profiles into known and unknown cell contents. <em>Commun Biol</em> <strong>3</strong>, 422 (2020). <a href="https://doi.org/10.1038/s42003-020-01146-2" class="external-link uri">https://doi.org/10.1038/s42003-020-01146-2</a>
</td>
</tr>
</tbody>
</table>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>You can install methyldeconv from <a href="https://github.com/" class="external-link">GitHub</a>, we recommend to use the <a href="https://github.com/r-lib/pak" class="external-link">pak</a> package manager:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># install the `pak` package manager</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"pak"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">pak</span><span class="fu">::</span><span class="fu"><a href="https://pak.r-lib.org/reference/pkg_install.html" class="external-link">pkg_install</a></span><span class="op">(</span><span class="st">"omnideconv/methyldeconv"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="example">Example<a class="anchor" aria-label="anchor" href="#example"></a>
</h2>
<p>methyldeconv can either be applied directly to a methylSet from the minfi package, or you can apply each method separately on a beta matrix with Illumina CpG IDs.</p>
<p>Both cases will be demonstrated here using example data from minfi:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/omnideconv/methyldeconv" class="external-link">methyldeconv</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/hansenlab/minfi" class="external-link">minfi</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">minfiData</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># use example data from Minfi</span></span>
<span><span class="va">methyl_set</span> <span class="op">&lt;-</span> <span class="fu">minfiData</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/minfiData/man/MsetEx.html" class="external-link">MsetEx</a></span></span>
<span><span class="va">ratio_set</span> <span class="op">&lt;-</span> <span class="fu">minfi</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/minfi/man/ratioConvert-methods.html" class="external-link">ratioConvert</a></span><span class="op">(</span><span class="va">methyl_set</span><span class="op">)</span></span>
<span><span class="va">beta_matrix</span> <span class="op">&lt;-</span> <span class="fu">minfi</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/minfi/man/MethylSet-class.html" class="external-link">getBeta</a></span><span class="op">(</span><span class="va">ratio_set</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="co"># run EpiDISH for deconvolution of example data</span></span>
<span><span class="va">result</span> <span class="op">&lt;-</span> <span class="fu">methyldeconv</span><span class="fu">::</span><span class="fu"><a href="reference/deconvolute.html">deconvolute</a></span><span class="op">(</span>methyl_set <span class="op">=</span> <span class="va">methyl_set</span>, method <span class="op">=</span> <span class="st">'epidish'</span><span class="op">)</span></span>
<span></span>
<span><span class="va">result_raw</span> <span class="op">&lt;-</span> <span class="fu">methyldeconv</span><span class="fu">::</span><span class="fu"><a href="reference/run_epidish.html">run_epidish</a></span><span class="op">(</span>beta_matrix <span class="op">=</span> <span class="va">beta_matrix</span>, mode<span class="op">=</span><span class="st">'RPC'</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="co"># you can also run multiple methods at the same time and get their results + aggregated results:</span></span>
<span><span class="va">result_multiple</span> <span class="op">&lt;-</span> <span class="fu">methyldeconv</span><span class="fu">::</span><span class="fu"><a href="reference/deconvolute_combined.html">deconvolute_combined</a></span><span class="op">(</span>methyl_set <span class="op">=</span> <span class="va">methyl_set</span>,</span>
<span>                                                      methods <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'epidish'</span>,<span class="st">'houseman'</span><span class="op">)</span>,</span>
<span>                                                      array <span class="op">=</span> <span class="st">'450k'</span><span class="op">)</span></span></code></pre></div>
<p>With methyldeconv you can either get the original result object of each respective method (<code>methyldeconv::run_XXX</code>) or a table with cell-type fractions that has a unified format between methods (<code><a href="reference/deconvolute.html">methyldeconv::deconvolute</a></code>).The unified results can be visualized using the methyldeconv functions <code><a href="reference/results_barplot.html">results_barplot()</a></code> or <code><a href="reference/results_boxplot.html">results_boxplot()</a></code>.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">methyldeconv</span><span class="fu">::</span><span class="fu"><a href="reference/results_barplot.html">results_barplot</a></span><span class="op">(</span><span class="va">result</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">methyldeconv</span><span class="fu">::</span><span class="fu"><a href="reference/results_boxplot.html">results_boxplot</a></span><span class="op">(</span><span class="va">result</span><span class="op">)</span></span></code></pre></div>
<p>Results from a run with more than one method (<code><a href="reference/deconvolute_combined.html">methyldeconv::deconvolute_combined</a></code>) can be easily visualized as well with <code>results_aggregated_boxplot</code>:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">methyldeconv</span><span class="fu">::</span><span class="fu"><a href="reference/results_aggregated_boxplot.html">results_aggregated_boxplot</a></span><span class="op">(</span><span class="va">result_multiple</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="dependencies">Dependencies<a class="anchor" aria-label="anchor" href="#dependencies"></a>
</h2>
<p>A full list of dependencies can be displayed with <code>pak</code>:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">pak</span><span class="fu">::</span><span class="fu"><a href="https://pak.r-lib.org/reference/pkg_deps_tree.html" class="external-link">pkg_deps_tree</a></span><span class="op">(</span><span class="st">"omnideconv/methyldeconv"</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
  </main><aside class="col-md-3"><div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/omnideconv/methyldeconv/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/omnideconv/methyldeconv/issues" class="external-link">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small>GPL (&gt;= 3)</small></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing methyldeconv</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Alexander Dietrich <br><small class="roles"> Author, maintainer </small>  </li>
</ul>
</div>

<div class="dev-status">
<h2 data-toc-skip>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/omnideconv/methyldeconv/actions/workflows/R-CMD-check.yml" class="external-link"><img src="https://github.com/omnideconv/methyldeconv/actions/workflows/R-CMD-check.yaml/badge.svg" alt="R-CMD-check"></a></li>
<li><a href="https://app.codecov.io/gh/omnideconv/methyldeconv?branch=main" class="external-link"><img src="https://codecov.io/gh/omnideconv/methyldeconv/branch/main/graph/badge.svg" alt="Codecov test coverage"></a></li>
</ul>
</div>

  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Alexander Dietrich.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
