<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Visualization Options in methyldeconv • methyldeconv</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Visualization Options in methyldeconv">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">methyldeconv</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.3</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/getting_started.html">Getting Started</a></li>
    <li><a class="dropdown-item" href="../articles/visualization.html">Visualization</a></li>
    <li><a class="dropdown-item" href="../articles/contribute.html">Contribute</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/omnideconv/methyldeconv/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Visualization Options in methyldeconv</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/omnideconv/methyldeconv/blob/main/vignettes/visualization.Rmd" class="external-link"><code>vignettes/visualization.Rmd</code></a></small>
      <div class="d-none name"><code>visualization.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="visualization-options-in-methyldeconv">Visualization Options in methyldeconv<a class="anchor" aria-label="anchor" href="#visualization-options-in-methyldeconv"></a>
</h2>
<p>methyldeconv provides several functions to visualize the results of
cell-type deconvolution analyses. This vignette demonstrates how to use
these visualization tools with example data.</p>
<div class="section level3">
<h3 id="example-data">Example Data<a class="anchor" aria-label="anchor" href="#example-data"></a>
</h3>
<p>We’ll use example data from the <code>minfi</code> and
<code>minfiData</code> packages:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/omnideconv/methyldeconv" class="external-link">methyldeconv</a></span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Loading required package: FlowSorted.Blood.450k</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: minfi</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: BiocGenerics</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: generics</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'generics'</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:base':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,</span></span>
<span><span class="co">##     setequal, union</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'BiocGenerics'</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:stats':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     IQR, mad, sd, var, xtabs</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:base':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     anyDuplicated, aperm, append, as.data.frame, basename, cbind,</span></span>
<span><span class="co">##     colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,</span></span>
<span><span class="co">##     get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,</span></span>
<span><span class="co">##     order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,</span></span>
<span><span class="co">##     rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,</span></span>
<span><span class="co">##     unsplit, which.max, which.min</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: GenomicRanges</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: stats4</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: S4Vectors</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'S4Vectors'</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:utils':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     findMatches</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:base':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     expand.grid, I, unname</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: IRanges</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: GenomeInfoDb</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: SummarizedExperiment</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: MatrixGenerics</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: matrixStats</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'MatrixGenerics'</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:matrixStats':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,</span></span>
<span><span class="co">##     colCounts, colCummaxs, colCummins, colCumprods, colCumsums,</span></span>
<span><span class="co">##     colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,</span></span>
<span><span class="co">##     colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,</span></span>
<span><span class="co">##     colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,</span></span>
<span><span class="co">##     colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,</span></span>
<span><span class="co">##     colWeightedMeans, colWeightedMedians, colWeightedSds,</span></span>
<span><span class="co">##     colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,</span></span>
<span><span class="co">##     rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,</span></span>
<span><span class="co">##     rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,</span></span>
<span><span class="co">##     rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,</span></span>
<span><span class="co">##     rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,</span></span>
<span><span class="co">##     rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,</span></span>
<span><span class="co">##     rowWeightedMads, rowWeightedMeans, rowWeightedMedians,</span></span>
<span><span class="co">##     rowWeightedSds, rowWeightedVars</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: Biobase</span></span></code></pre>
<pre><code><span><span class="co">## Welcome to Bioconductor</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     Vignettes contain introductory material; view with</span></span>
<span><span class="co">##     'browseVignettes()'. To cite Bioconductor, see</span></span>
<span><span class="co">##     'citation("Biobase")', and for packages 'citation("pkgname")'.</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'Biobase'</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:MatrixGenerics':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     rowMedians</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:matrixStats':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     anyMissing, rowMedians</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: Biostrings</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: XVector</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'Biostrings'</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:base':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     strsplit</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: bumphunter</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: foreach</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: iterators</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: parallel</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: locfit</span></span></code></pre>
<pre><code><span><span class="co">## locfit 1.5-9.12   2025-03-05</span></span></code></pre>
<pre><code><span><span class="co">## Setting options('download.file.method.GEOquery'='auto')</span></span></code></pre>
<pre><code><span><span class="co">## Setting options('GEOquery.inmemory.gpl'=FALSE)</span></span></code></pre>
<pre><code><span><span class="co">## → checking methyldeconv environment and dependencies</span></span></code></pre>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/hansenlab/minfi" class="external-link">minfi</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">minfiData</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Loading required package: IlluminaHumanMethylation450kmanifest</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: IlluminaHumanMethylation450kanno.ilmn12.hg19</span></span></code></pre>
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Example data</span></span>
<span><span class="va">methyl_set</span> <span class="op">&lt;-</span> <span class="fu">minfiData</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/minfiData/man/MsetEx.html" class="external-link">MsetEx</a></span></span>
<span><span class="va">ratio_set</span> <span class="op">&lt;-</span> <span class="fu">minfi</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/minfi/man/ratioConvert-methods.html" class="external-link">ratioConvert</a></span><span class="op">(</span><span class="va">methyl_set</span><span class="op">)</span></span>
<span><span class="va">beta_matrix</span> <span class="op">&lt;-</span> <span class="fu">minfi</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/minfi/man/MethylSet-class.html" class="external-link">getBeta</a></span><span class="op">(</span><span class="va">ratio_set</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Run deconvolution</span></span>
<span><span class="va">result</span> <span class="op">&lt;-</span> <span class="fu">methyldeconv</span><span class="fu">::</span><span class="fu"><a href="../reference/deconvolute.html">deconvolute</a></span><span class="op">(</span>methyl_set <span class="op">=</span> <span class="va">methyl_set</span>, method <span class="op">=</span> <span class="st">'epidish'</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning in check_input_beta(beta_matrix): 12 NA values detected in your beta</span></span>
<span><span class="co">## matrix. Replacing them with 0.5.</span></span></code></pre>
<pre><code><span><span class="co">## RPC was chosen as default for "mode"</span></span></code></pre>
<pre><code><span><span class="co">## blood was chosen as default for "reference"</span></span></code></pre>
<pre><code><span><span class="co">## Starting EpiDISH deconvolution with mode RPC ...</span></span></code></pre>
<div class="sourceCode" id="cb50"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">result_multiple</span> <span class="op">&lt;-</span> <span class="fu">methyldeconv</span><span class="fu">::</span><span class="fu"><a href="../reference/deconvolute_combined.html">deconvolute_combined</a></span><span class="op">(</span>methyl_set <span class="op">=</span> <span class="va">methyl_set</span>,</span>
<span>                                                      methods <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'epidish'</span>,<span class="st">'houseman'</span><span class="op">)</span>,</span>
<span>                                                      array <span class="op">=</span> <span class="st">'450k'</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning in check_input_beta(beta_matrix): 12 NA values detected in your beta</span></span>
<span><span class="co">## matrix. Replacing them with 0.5.</span></span></code></pre>
<pre><code><span><span class="co">## RPC was chosen as default for "mode"</span></span></code></pre>
<pre><code><span><span class="co">## blood was chosen as default for "reference"</span></span></code></pre>
<pre><code><span><span class="co">## Starting EpiDISH deconvolution with mode RPC ...</span></span></code></pre>
<pre><code><span><span class="co">## 450k was chosen as default for "array"</span></span></code></pre>
<pre><code><span><span class="co">## Blood was chosen as default for "compositeCellType"</span></span></code></pre>
<pre><code><span><span class="co">## IlluminaHumanMethylationEPIC was chosen as default for "referencePlatform"</span></span></code></pre>
<pre><code><span><span class="co">## IDOL was chosen as default for "probeSelect"</span></span></code></pre>
<pre><code><span><span class="co">## [estimateCellCounts2] The function will assume that no preprocessing has been performed. Using 'preprocessQuantile' in prenormalized data is experimental and it should only be run under the user responsibility</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: FlowSorted.Blood.EPIC</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: ExperimentHub</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: AnnotationHub</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: BiocFileCache</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: dbplyr</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'AnnotationHub'</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:Biobase':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     cache</span></span></code></pre>
<pre><code><span><span class="co">## see ?FlowSorted.Blood.EPIC and browseVignettes('FlowSorted.Blood.EPIC') for documentation</span></span></code></pre>
<pre><code><span><span class="co">## loading from cache</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: IlluminaHumanMethylationEPICmanifest</span></span></code></pre>
<pre><code><span><span class="co">## [convertArray] Casting as IlluminaHumanMethylationEPIC</span></span></code></pre>
<pre><code><span><span class="co">## [estimateCellCounts2] Combining user data with reference (flow sorted) data.</span></span></code></pre>
<pre><code><span><span class="co">## Warning in asMethod(object): NAs introduced by coercion</span></span></code></pre>
<pre><code><span><span class="co">## [estimateCellCounts2] Processing user and reference data together.</span></span></code></pre>
<pre><code><span><span class="co">## [preprocessQuantile] Mapping to genome.</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: IlluminaHumanMethylationEPICanno.ilm10b4.hg19</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'IlluminaHumanMethylationEPICanno.ilm10b4.hg19'</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:IlluminaHumanMethylation450kanno.ilmn12.hg19':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     Islands.UCSC, Locations, Manifest, Other, SNPs.132CommonSingle,</span></span>
<span><span class="co">##     SNPs.135CommonSingle, SNPs.137CommonSingle, SNPs.138CommonSingle,</span></span>
<span><span class="co">##     SNPs.141CommonSingle, SNPs.142CommonSingle, SNPs.144CommonSingle,</span></span>
<span><span class="co">##     SNPs.146CommonSingle, SNPs.147CommonSingle, SNPs.Illumina</span></span></code></pre>
<pre><code><span><span class="co">## [preprocessQuantile] Fixing outliers.</span></span></code></pre>
<pre><code><span><span class="co">## [preprocessQuantile] Quantile normalizing.</span></span></code></pre>
<pre><code><span><span class="co">## [estimateCellCounts2] Using IDOL L-DMR probes for composition estimation.</span></span></code></pre>
<pre><code><span><span class="co">## [estimateCellCounts2] Estimating proportion composition (prop), if you provide cellcounts those will be provided as counts in the composition estimation.</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="barplot-of-deconvolution-results">Barplot of Deconvolution Results<a class="anchor" aria-label="anchor" href="#barplot-of-deconvolution-results"></a>
</h3>
<p>The <code><a href="../reference/results_barplot.html">results_barplot()</a></code> function creates a barplot for
each sample, showing the estimated cell-type fractions:</p>
<div class="sourceCode" id="cb82"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">methyldeconv</span><span class="fu">::</span><span class="fu"><a href="../reference/results_barplot.html">results_barplot</a></span><span class="op">(</span><span class="va">result</span><span class="op">)</span></span></code></pre></div>
<p><img src="visualization_files/figure-html/unnamed-chunk-2-1.png" width="768"></p>
</div>
<div class="section level3">
<h3 id="boxplot-of-deconvolution-results">Boxplot of Deconvolution Results<a class="anchor" aria-label="anchor" href="#boxplot-of-deconvolution-results"></a>
</h3>
<p>The <code><a href="../reference/results_boxplot.html">results_boxplot()</a></code> function creates a boxplot for
each cell type, summarizing the distribution of estimated fractions
across samples:</p>
<div class="sourceCode" id="cb83"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">methyldeconv</span><span class="fu">::</span><span class="fu"><a href="../reference/results_boxplot.html">results_boxplot</a></span><span class="op">(</span><span class="va">result</span><span class="op">)</span></span></code></pre></div>
<p><img src="visualization_files/figure-html/unnamed-chunk-3-1.png" width="768"></p>
</div>
<div class="section level3">
<h3 id="aggregated-boxplot-for-multiple-methods">Aggregated Boxplot for Multiple Methods<a class="anchor" aria-label="anchor" href="#aggregated-boxplot-for-multiple-methods"></a>
</h3>
<p>If you run multiple methods using
<code><a href="../reference/deconvolute_combined.html">deconvolute_combined()</a></code>, you can visualize the aggregated
results with <code><a href="../reference/results_aggregated_boxplot.html">results_aggregated_boxplot()</a></code>:</p>
<div class="sourceCode" id="cb84"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">methyldeconv</span><span class="fu">::</span><span class="fu"><a href="../reference/results_aggregated_boxplot.html">results_aggregated_boxplot</a></span><span class="op">(</span><span class="va">result_multiple</span><span class="op">)</span></span></code></pre></div>
<p><img src="visualization_files/figure-html/unnamed-chunk-4-1.png" width="768"></p>
<hr>
<p>These visualization functions help you interpret and compare the
results of different deconvolution methods. For more customization
options, see the function documentation or explore the source code in
the <code>R/visualization.R</code> file.</p>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Alexander Dietrich.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer>
</div>





  </body>
</html>
